postgres-lock/src/main/java/com/netflix/conductor/postgreslock/lock/PostgresLock.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2024 Conductor Authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.netflix.conductor.postgreslock.lock;
+
+import java.sql.Timestamp;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import javax.sql.DataSource;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.netflix.conductor.core.sync.Lock;
+
+public class PostgresLock implements Lock {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final String lockOwnerId;
+
+    public PostgresLock(DataSource dataSource) {
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
+        this.lockOwnerId = UUID.randomUUID() + "-" + System.currentTimeMillis();
```

Comment:
I have concerns about this part. PostgresLock is instantiated once per Conductor instance, the lockOwnerId is shared over this instance. A single instance of Conductor has multiple concurrent routines which each could fight over the lock, since each of them share the same lockOwnerId, they could be taking over each other's locks.